### DISCUSSION TITLE
ModifyHitdef state controller

### DISCUSSION THREAD

**Discussion by potsmugen** [ikemen-engine/Ikemen-GO#1229] [Feature Request]
### Is your feature request related to a problem? Please describe.

Currently, if you want to change something about an active Hitdef, you have to make another one on top of it. Since Hitdefs are the longest state controllers, this can create a lot of redundancy.

Another issue with Hitdefs (and the engine in general) is that string fields can't accept ifelse expressions, so for example to have one Hitdef with conditional Hitflags, you have to duplicate it.

### Describe the solution you'd like

A ModifyHitdef sctrl. The name is fairly self-explanatory, but this would modify (only) the specified parameters in the _active_ Hitdef. Very similar to ModifyExplod. It would do the exact same thing to a ReversalDef.

The difference between modifying a Hitdef and creating a new one is that, for instance, if you're modifying a Hitdef that has already hit one player, it won't hit the same player again.

One use case I had in mind was something like this:

```
if map(jugglepoints) > 20 {
        modifyHitdef{hitflag: MA} # disable F
}
```

So basically if the (custom) juggle points were above 20, the Hitdef would lose the F hitflag, becoming unable to juggle.

**Comment by Lazin3ss** [comment]
This is a great quality of life improving feature. That being said, it sounds like it would warrant some hitdef logic rewriting, considering the nature of hitdef and gethitvar objects.

**Comment by potsmugen** [comment]
I was hoping it'd be simple, like what happens when you create a new Hitdef: read parameter -> apply parameter. Like running the Hitdef "runSub" function again. If it involves such a rewrite it's probably not worth it.

**Comment by ghost** [comment]
As a fan of sad claps, this would be extra fun to play around with!

**Comment by potsmugen** [comment]
Implemented with 04b8417
